### PR TITLE
StompJS: Fix errorCallback signature

### PR DESCRIPTION
**Pull request by MaximeAnsquer**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
http://stomp.github.com/stomp-specification-1.1.html#ERROR
https://github.com/jmesnil/stomp-websocket/blob/master/src/stomp.coffee#L312

### CODE REVIEW

**Review by JimiC — CHANGES_REQUESTED** [review]
According to https://github.com/jmesnil/stomp-websocket/blob/master/src/stomp.coffee#L321 it can also be of `string`.

**Comment by MaximeAnsquer** [comment]
@JimiC What do you think about `errorCallback?: (error: Frame|string) => any` ?
And by the way, could we not change the return type from `any` to `void` ?

**Comment by JimiC** [comment]
`errorCallback?: (error: Frame | string) => any` is the correct typing.

I'm guessing you are asking about changing the return type of the `errorCallback` function, right? If that's the case, `any` is the safest type to use. You can not be certain if there will be a use case where the author needs to return something from the `errorCallback` function. The library does not explicitly specifies the return type so it will be risky to use `void` as it may break things.

File: types/stompjs/index.d.ts [hunk 1]
```diff
@@ -26,9 +26,9 @@ export class Client {
 
     debug(...args: string[]): any;
 
-    connect(headers: { login: string, passcode: string, host?: string }, connectCallback: (frame?: Frame) => any, errorCallback?: (error: string) => any): any;
-    connect(headers: { }, connectCallback: (frame?: Frame) => any, errorCallback?: (error: string) => any): any;
-    connect(login: string, passcode: string, connectCallback: (frame?: Frame) => any, errorCallback?: (error: string) => any, host?: string): any;
+    connect(headers: { login: string, passcode: string, host?: string }, connectCallback: (frame?: Frame | string) => any, errorCallback?: (error: Frame) => any): any;
```
Comment (JimiC) [inline types/stompjs/index.d.ts]: You changed the wrong function

Comment (MaximeAnsquer) [reply types/stompjs/index.d.ts]: Woops!

**Review by JimiC — APPROVED** [review]

**Pull request merged**